cue/spec.cue

```diff
@@ -0,0 +1,258 @@
+package spec 
+
+import (
+    "time"
+)
+
+// TODO: consider adding defaults to more fields
```

Comment:
It is definitely worthwhile to capture all of the defaults in this PR. We actually have extensive defaults, mostly undocumented.